src/app/(auth)/sign-in/components/SignInFormContainer.tsx

```diff
@@ -38,10 +56,30 @@ function SignInFormContainer({
     resolver: zodResolver(validationSchema),
   });
 
-  const onSubmit: SubmitHandler<ValidationSchema> = (data) => {
-    console.log(data);
+  const onSubmit: SubmitHandler<ValidationSchema> = async (data) => {
+    const { email, password } = data;
+    const [res, error] = await serverSignInAction({ email, password });
+
+    if (res) {
+      dispatch(clientSignIn());
+      router.push("/dashboard");
```

Comment:
use routePaths here. Also use router.replace instead, I think it's slightly better usecase here.